doc: Initial NVMe-oF gateway docs (#55540)
ceph/ceph: doc/rbd/nvmeof-overview.rst

.. _ceph-nvmeof:

======================
 Ceph NVMe-oF Gateway
======================

The NVMe-oF Gateway presents an NVMe-oF target that exports

> RADOS Block Device (RBD) images as NVMe namespaces. The NVMe/-oF protocol allows

Reviewer: ```suggestion
RADOS Block Device (RBD) images as NVMe namespaces. The NVMe-oF protocol allows
```

Reviewer 2: Remedied.

> clients (Initiators) to send NVMe commands to storage devices (targets) over a

Reviewer: ```suggestion
clients (initiators) to send NVMe commands to storage devices (targets) over a
```

Reviewer 2: Accepted.

TCP/IP network, enabling clients without native Ceph client support to access
Ceph block storage.  

> Each NMVe-oF gateway consits of an `SPDK <https://spdk.io/>`_ NVMe-oF target

Reviewer: ```suggestion
Each NVMe-oF gateway consists of an `SPDK <https://spdk.io/>`_ NVMe-oF target
```

Reviewer 2: Fixed.

Reviewer: Not quite -- "NMVe" is still there.

Reviewer 2: I have removed it, at long last.
